src/Reflection/ReflectionClass.php

```diff
@@ -519,7 +519,7 @@ static function (ReflectionClassConstant $constant) : string {
      * Get an associative array of the defined constants in this class,
      * with keys as constant names and values as {@see ReflectionClassConstant} objects.
      *
-     * @return ReflectionClassConstant[] indexed by name
+     * @return array<string, ReflectionClassConstant> indexed by name
```

Comment:
This raise a new error on Psalm too because their internal stubs expects this to be an array<int, ReflectionCl
assConstant>. Not sure who's right on this

Reply:
I think the stub may need updating here

Reply:
https://3v4l.org/t05Dv I took this example directly from php.net. The index of array is numerical.
I'll add the case to #530 as it is roughly the same issue

Reply:
Yeh, the question is whether it is a list or not: I think it is a list in core too

Reply:
> Yeh, the question is whether it is a list or not: I think it is a list in core too

I think you got me wrong. The library returns `array<string, ReflectionClassConstant>` right now but native Object return `array<int, ReflectionClassConstant>` (and more probably a `list<ReflectionClassConstant>` but this is not the issue).

I'll silence those errors in psalm to allow the merge, but I think those are legitimate issues.

Reply:
Oh, careful! We have adapters and then there is this stuff: the adapters are supposed to be compliant with internal definitions, but these classes don't need to be.

Reply:
> Oh, careful! We have adapters and then there is this stuff: the adapters are supposed to be compliant with internal definitions, but these classes don't need to be.

oh, I get it. But Psalm sees that this class implements Reflection and warn about the type of the interface being different than declared